File: nim/loop.nim
# Compiled with v1.0.4 using :
# nim compile -d:release --passC:-flto --passL:-s --gc:markAndSweep --out:loop.exe loop.nim

var a: int = 10
var n: int
var i: int
var result: int

for n in 1..10000000:
    result = 1
    for i in 1..a:
        result = result + (a - i) + n
    # End for
# End for

echo result

File: Rust/loop.rs
// Rust 1.4
// compiled with: rustc -O loop.rs

fn main() {
	let a: u32 = 10;
	let mut result: u32 = 1;

	for n in 1..10000001 {
		result = 1;
		for i in 1..(a+1) {
			result = result + (a - i) + n;
		}
	}

	println!("{}", result);
}